tests/unit/heap.c

```diff
@@ -23,9 +23,66 @@
 
 #include <igraph.h>
 
+#include "test_utilities.inc"
+
 int main() {
 
-    /* This is not used by any functions any more,
-       no need to test it right now */
+    igraph_heap_t h;
+    igraph_integer_t i;
+    igraph_real_t *arr =(igraph_real_t*) calloc(5, sizeof(igraph_real_t));
+
+    /* igraph_heap_init, igraph_heap_destroy  */
+    printf("Test igraph_heap_init, igraph_heap_destroy\n");
+    igraph_heap_init(&h, 0);
+    igraph_heap_destroy(&h);
+    igraph_heap_init(&h, 10);
+    igraph_heap_destroy(&h);
+
+    /* igraph_heap_reserve */
+    printf("Test igraph_heap_reserve\n");
+    igraph_heap_init(&h, 0);
+    igraph_heap_reserve(&h, 10);
+    igraph_heap_reserve(&h, 5);
+    igraph_heap_destroy(&h);
+
+    /* igraph_heap_init_array*/
+    printf("Test igraph_heap_init_array\n");
+    igraph_heap_init_array(&h,arr,5);
```

Comment:
It would be nice to test that this actually copies the elements from `arr`. Therefore, I would use an array with known elements, such as `random_list` below, then check that the correct elements were added to the heap.